tokio/src/io/util/read_to_end.rs

```diff
@@ -6,12 +6,14 @@ use std::mem::MaybeUninit;
 use std::pin::Pin;
 use std::task::{Context, Poll};
 
-#[derive(Debug)]
-#[must_use = "futures do nothing unless you `.await` or poll them"]
-pub struct ReadToEnd<'a, R: ?Sized> {
-    reader: &'a mut R,
-    buf: &'a mut Vec<u8>,
-    start_len: usize,
+cfg_io_util! {
+    #[derive(Debug)]
+    #[must_use = "futures do nothing unless you `.await` or poll them"]
+    pub struct ReadToEnd<'a, R: ?Sized> {
```

Comment:
As `ReadToEnd` is used internally even when `io-util` feature is disabled, it seems to be necessary to use an attribute.

Reply:
`ReadToEnd` is only **publicly* available w/ the feature. I do need to fix it though.